Functions/Import/xASL_imp_StudyParPriority.m

```diff
@@ -0,0 +1,90 @@
+function studyParSpecific = xASL_imp_StudyParPriority(studyParFull, subjectName, sessionName, runName)
+%xASL_imp_StudyParPriority Takes the studyParFull a prioritizes based on the current subject/session/run
+%
+% FORMAT: studyParSpecific = xASL_imp_StudyParPriority(studyParFull[, subjectName, sessionName, runName])
+%
+% INPUT:
+%   studyParFull - StudyPar possibly containing several contexts (REQUIRED, STRUCT)
+%   subjectName  - Name of the current subject (OPTIONAL, DEFAULT = '')
+%   sessionName  - Name of the current session (OPTIONAL, DEFAULT = '')
+%   runName      - Name of the current run (OPTIONAL, DEFAULT = '')
+%
+% OUTPUT:
+%   studyParSpecific - Resolved studyPar for a specific subject/session/run
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION:    Takes studyPar with possibly several contexts and resolves the priority and saves the individual studyPar.
```

Comment:
That's the whole point - it's a priority list and not inheritance. Therefore, inherently, it will overwrite things and that's the whole point - that things are defined broadly, and then you proceed to specify on session level, visit level or subject and more specific left and overwrite at each time - that's why we have the priorities and that's why it is called like that.

I used the word inheritance while explaining just to make it easier to understand. 

So conflict are dealt with by overwriting by design and there's no point in reporting that. 
Of course we can try to detect clearly duplicity entries like you have, but the point is that you ARE overwriting.

Reply:
So inheritance from top to bottom, where each new layer overwrites any conflict with the previous layer?

This is really complicated for the user. Again, the Table ... we need it now. This is asking for trouble.
I'm fine merging this now, on the condition that this doesn't appear in a release without the Table.
Let users (including our PhD students!) use the Table.

Just to explain to you: imagine that you get the same salary throughout the year, but there are a few exceptions in which you get reimbursements, and halfway through the year you get a salary raise. Even though you are right that you can describe this easier, you still want to check on your account that you actually received your salary; your will check this month by month right? ;) That's the same with this case.

So yes, your JSON solution is helpful in a simple case, where each participant has an identical number of sessions and the second session always has a different PLD than the first. But reality is unfortunately different:
A) one of the sessions failed so they repeat it; so a few subjects have 3 sessions
B) most of the subjects have 2 visits where the second visit has a PASL and the first a PCASL; but because the scanner was replaced in the middle, this is not consistent. Some subjects also had PASL on baseline.
And I can go on and on; I have seen this many times. We need the Table. Tables are human readable.

Reply:
JSON is also easier in this case. Cause you define the general case with low priority and more exceptional cases are redefined when needed. There's no scenario that would have been faster and less error-prone to describe with a table ;)

sorry, but your examples are not good. If checking the salary, I would check - for the whole year, I've received XX EUR, for two months YY EUR.

Just try to imagine how difficult will be to fill in the table. Copy pasting the content 1000 times for 1000 subjects. then trying to find where you made a mistake :) Then you add two more subjects and you need to repeat this whole again :)